pkg/daemon/update.go

```diff
@@ -375,40 +375,39 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 				return err
 			}
 		}
-	}
-
-	if dn.nodeWriter != nil {
-		var nodeName string
-		var nodeObjRef corev1.ObjectReference
-		if dn.node != nil {
-			nodeName = dn.node.ObjectMeta.GetName()
-			nodeObjRef = corev1.ObjectReference{
-				Kind: "Node",
-				Name: dn.node.GetName(),
-				UID:  dn.node.GetUID(),
+		if dn.nodeWriter != nil {
+			var nodeName string
+			var nodeObjRef corev1.ObjectReference
+			if dn.node != nil {
+				nodeName = dn.node.ObjectMeta.GetName()
+				nodeObjRef = corev1.ObjectReference{
+					Kind: "Node",
+					Name: dn.node.GetName(),
+					UID:  dn.node.GetUID(),
+				}
+			}
+			// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+			// Remove this when we can ensure all events are sent before exiting.
+			t := metav1.NewTime(time.Now())
```

Comment:
the way the two events are created are still different though? is that ok?

Reply:
We'll try to make that consistent later. For now, this should make it such that on each upgrade, the event only shows up once each